_posts/2019-09-13-dns-tips.md

```diff
@@ -0,0 +1,101 @@
+| Helper    | ocp4-helper.local.lan    | 10.0.0.150 | DNS/HTTPD & Load Balancer                     |
+
+The domain is: local.lan
+And this setup has a cluster of 3 masters, 1 worker, the bootstrap server and a helper server running the DNS, HTTPD and a Load Balancer.
+It's an example, you can configure it as you wish.
+
+## CoreDNS configuration files
+The following Corefile is an example for this configuration:
+```
+Corefile 
+.:53 {
+    log
+    errors
+    forward . 8.8.8.8
+}
+
+local.lan:53 {
+    log
+    errors
+    file /etc/coredns/db.local.lan
+}
+
+```
+
+And the db.local.lan looks like the following snippet:
+```
+$ORIGIN local.lan.
+$TTL 10800      ; 3 hours
+@       3600 IN SOA sns.dns.icann.org. noc.dns.icann.org. (
+                                2019091301 ; serial
+                                7200       ; refresh (2 hours)
+                                3600       ; retry (1 hour)
+                                1209600    ; expire (2 weeks)
+                                3600       ; minimum (1 hour)
+                                )
+
+_etcd-server-ssl._tcp.ocp4.local.lan. 8640 IN    SRV 0 10 2380 etcd-0.ocp4.local.lan.
+_etcd-server-ssl._tcp.ocp4.local.lan. 8640 IN    SRV 0 10 2380 etcd-1.ocp4.local.lan.
+_etcd-server-ssl._tcp.ocp4.local.lan. 8640 IN    SRV 0 10 2380 etcd-2.ocp4.local.lan.
+
+api.ocp4.local.lan.                        A                10.0.0.150
+api-int.ocp4.local.lan.                    A                10.0.0.150
+ocp4-master-0.local.lan.                   A                10.0.0.100
+ocp4-master-1.local.lan.                   A                10.0.0.101
+ocp4-master-2.local.lan.                   A                10.0.0.102
+ocp4-worker-0.local.lan.                   A                10.0.0.200
+ocp4-bootstrap.local.lan.                  A                10.0.0.99
+etcd-0.ocp4.local.lan.                     IN  CNAME ocp4-master-0.local.lan.
+etcd-1.ocp4.local.lan.                     IN  CNAME ocp4-worker-1.local.lan.
+etcd-2.ocp4.local.lan.                     IN  CNAME ocp4-worker-2.local.lan.
+
+$ORIGIN apps.ocp4.local.lan.
+*                                                    A                10.0.0.150
+
+```
+
+Once the CoreDNS is configured and running (in a container or natively) the installation can be started.
+
+## Warning
+
+It's importat tho advise that in this example the servers defined in the DNS are the ones really used for installing the initial cluster, 
+in the case you may want to add future masters be careful with the dns registers for the etc servers and don't define them until you really need them.
+Otherwise, while you are running the bootstrap phase of your installation, the etcd cluster will autodiscover each '_etcd-server-ssl._tcp.ocp4.local.bin' server 
+and will try to connect to each of them and in the case there is one that can not be reached, the cluster won't start failing the installation.
```

Comment:
in the case one can not be reached, the cluster will fail to esbablish

Reply:
Changed, thx,